### ISSUE TITLE
Cancelling consumer by consumer tag not removing the tag from queue

### ISSUE THREAD

**Issue by tezhm** [php-amqp/php-amqp#385]
Test to demonstrate:

```php
    $queue = new AMQPQueue($this->channel);
    $queue->declareQueue();

    // Asynchronously consuming
    $queue->consume(null, AMQP_NOPARAM, 'someid');

    // Showing that current consumer tag is from last consume call
    $this->assertEquals('someid', $queue->getConsumerTag());

    // Cancel by consumer tag -> expecting the current consumer tag to clear
    $queue->cancel('someid');

    // This fails as consumer tag is still 'someid'
    $this->assertNull($queue->getConsumerTag());
```

**Comment by lstrojny** [comment]
Please open a pull request with a failing test

**Comment by tezhm** [comment]
@lstrojny added a pull request with failing test

**Comment by lstrojny** [comment]
Closed in #432
